terraform.md

```diff
@@ -0,0 +1,120 @@
+# Terraform
+
+Terraform is a really powerful tool with a lot of freedom in the way you use it.
+It isn't a very opinionated tool, however if you don't stick with rules it can
+easily get complicated and hard to maintain.
+
+This is why we decided to write our own guideline on how to structure a
+`terraform` codebase.
+
+## File Structure and Filenames
+
+Files should be named with the AWS (or other cloud provider) components they are
+referring to. Keeping this open to future needs, if a component is getting more
+in use than others we can make the filename more specific.
+The idea is to keep a balance between too many small files and too little huge
+files.
+
+```
+/my-projet/
+├─ modules/
+│   └─ database/
+│       ├─ ec2.tf
+│       ├─ cloudwatch.tf
+│       ├─ outputs.tf
+│       ├─ rds.tf
+│       └─ variables.tf
+├─ backend.tf        # To set the backend instruction of terraform
+├─ cloudflare.tf
+├─ config.tf         # To set your provider(s), required terraform version, ...)
+├─ ec2_api.tf        # Subset of EC2 for API related resources.
+├─ ec2_worker.tf     # Subset of EC2 for workers related resources.
+├─ rds.tf
+├─ keypair.tf
+├─ iam.tf
+├─ variables.tf
+└─ outputs.tf
+```
+
+__As filenames doesn't matter for terraform, we can easily refactor them
+overtime.__
+
+## Patterns
+
+### Database Security Group
```

Comment:
Better to call this "Allow callers to add security group ingress rules" or something similar that describes the pattern.